skimage/segmentation/manual_segmentation.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.patches import Polygon
+from matplotlib.collections import PatchCollection
+
+
+def manual(image, speed_up=1):
+    """Return a binary image based on the selections made with mouse clicks.
+
+    Parameters
+    ----------
+    image : (M, 2) array
+        Grayscale or RGB 2D image.
+
+    speed_up : int
+        Skips vertices in integer steps to speed up mask generation. This
+        must be a non-zero integer.
+
+    Returns
+    -------
+    mask : (M, 2) binary array
+        Segmented binary mask.
+
+    Notes
+    -----
+    Use the cursor to draw objects. Increasing speed_up value will 
+    generate the mask faster, but the objects will be less smoother. 
+
+    Examples
+    --------
+    >>> from skimage import data, segmentation
+    >>> camera = data.camera()
+    >>> mask = segmentation.manual(camera)
+    >>> # Use the cursor to draw objects
+    >>> io.imshow(mask)
+    >>> io.show()
+
+    """
+    image = np.squeeze(image)
+
+    if image.ndim not in (2, 3):
+        raise TypeError('Only 2-D images or 3-D images supported.')
+
+    manual.list_of_verts = []
+    manual.polygons_selection = []
+    manual.polygon_list = []
+
+    _select_lasso(image)
+
+    if image.ndim is 3:
+        image = np.squeeze(image[:, :, :1])
+
+    mask = np.zeros(image.shape)
+
+    yshape, xshape = image.shape
+    y_grid, x_grid = np.mgrid[:yshape, :xshape]
+    all_pixels = np.vstack((x_grid.ravel(), y_grid.ravel())).T
+
+    for verts in manual.list_of_verts:
+        verts_ = verts[::speed_up]
```

Comment:
I definitely don't like this way of speeding things up. Users expect that their clicks will have an effect. Objects containing "spiky" shapes will definitely fail with this approach.

Maybe try using [`skimage.draw.polygon`](http://scikit-image.org/docs/dev/api/skimage.draw.html#polygon) instead of `matplotlib.path.Path`? I don't know whether it'll be faster but it's worth a shot.